### ISSUE TITLE
Issues with Scala 3 enums derived from a trait

### ISSUE THREAD

**Issue by OndrejSpanel** [sirthias/borer#695]
Following code gives error with borer 1.14.0 and Scala 3.3.3:

```scala
package com.github.ondrejspanel.borer

import io.bullet.borer._
import io.bullet.borer.derivation.MapBasedCodecs._

sealed trait EBase
case class EE() extends EBase

enum E extends EBase:
  case A
  case B

object E:
  given codec: Codec[E] = deriveCodec

object EBase:
  given codec: Codec[EBase] = deriveAllCodecs

@main
def main(): Unit = {
  val c: EBase = EE()
  println(Json.encode(c).toUtf8String)
}
```

Error is:

> value E is not a member of object com.github.ondrejspanel.borer.EBase
>   given codec: Codec[EBase] = deriveAllCodecs

When removing EE, a different error is shown:

> Could not find any sub-types of `EBase`, likely because `EBase` is not a fully closed ADT.
> Do you maybe have a `sealed trait` somewhere, which has no subclasses?

**Comment by sirthias** [comment]
Ok, interesting, thank you @OndrejSpanel !
I'll look into this ASAP.

**Comment by mrdziuban** [comment]
I also just ran into the second error, `... is not a fully closed ADT`. Would love to see support for `enum`s!

**Comment by sirthias** [comment]
Ok, thank you, @mrdziuban !
The bug is confirmed but unfortunately not super easy to fix.
Still, it needs to be corrected and will be done.

Note that `Enum`s are supported already, the problem only arises of they extend another `trait` or `class`.

**Comment by mrdziuban** [comment]
Thanks @sirthias, I realized why it wasn't working for me, my issue is unrelated to extending another `trait` or `class`. It looks like using `Encoder.All.derived` or `deriveAllEncoders` doesn't work when the `enum` has only `case object` members -- https://scastie.scala-lang.org/EB8BVPG0QMGkeqCSg2Pjlg

Is this intentional? I would like to use `Encoder.All.derived` for all of my `enum`s regardless of whether they're all `case objects` or some `case class`es are mixed in. `sealed trait`s and `sealed abstract class`es do not have the same issue.

<img width="626" alt="image" src="https://github.com/sirthias/borer/assets/4718399/9b8f8dc9-d5c0-4fe4-bd0c-362b87868446">

**Comment by sirthias** [comment]
@mrdziuban Ah, I see.
The case that an enum doesn't define any actual sub-classes (but only simple cases) is already supported, but the error message you are seeing isn't good. I've just pushed a fix for that.
In your case everything will work as expected with a simple `derives Encoder` or `deriveEncoder` call.

Background: Simple Enum cases do _not_ actually define any new types or sub-classes. They simply designate _instances_ of the enum type itself. Therefore a single `Encoder`, `Decoder` or `Codec` is enough to support all cases and you don't need the `deriveAll` macro at all.

**Comment by sirthias** [comment]
I just published release 1.14.1 containing the full fill for this issue.
Thank you again for reporting!
